index.js

```diff
@@ -70,11 +92,35 @@ module.exports = {
     });
 
     var bodyParser = require('body-parser');
-    app.use('/csp-report', bodyParser.json({type:'application/csp-report'}));
-    app.use('/csp-report', bodyParser.json({type:'application/json'}));
-    app.use('/csp-report', function(req, res, next) {
+    app.use(REPORT_PATH, bodyParser.json({ type: 'application/csp-report' }));
+    app.use(REPORT_PATH, bodyParser.json({ type: 'application/json' }));
+    app.use(REPORT_PATH, function(req, res, next) {
       console.log(chalk.red('Content Security Policy violation:') + '\n\n' + JSON.stringify(req.body, null, 2));
-      res.send({status:'ok'});
+      res.send({ status:'ok' });
     });
+  },
+
+  contentFor: function(type, appConfig) {
+    if ((type === 'head' && appConfig.contentSecurityPolicyMeta)) {
+      var policyObject = appConfig.contentSecurityPolicy;
+      var policyHeader = appConfig.contentSecurityPolicyHeader;
+      var policyString = buildPolicyString(policyObject);
+
+      unsupportedDirectives(policyObject).forEach(function(name) {
+        var msg = 'CSP deliverd via meta does not support `' + name + '`, ' +
+                  'per the W3C recommendation.';
+        console.log(chalk.yellow(msg));
+      });
+
+      if (!policyHeader || policyHeader !== CSP_HEADER) {
```

Comment:
I don't fully understand this conditional. I thought the meta scenario was for when you couldn't set headers? Why is the header required?